src/validation.cpp

```diff
@@ -3818,7 +3821,7 @@ bool ProcessNewBlock(const CChainParams& chainparams, const std::shared_ptr<cons
     NotifyHeaderTip();
 
     BlockValidationState state; // Only used to report errors, not invalidity - ignore it
-    if (!::ChainstateActive().ActivateBestChain(state, chainparams, pblock))
+    if (!ActivateBestChain(state, chainparams, pblock))
```

Comment:
In commit "protect g_chainman with cs_main" (d88f75e495a46ae474d8ad67faaec13ef1950f5e)

I don't understand the two swaps between `::ActivateBestChain(...)` and `::ChainstateActive().ActivateBestChain(...)` in this commit. They don't seem to do anything, and if I revert them there don't seem to be obvious problems.

Not specific to this PR but generally it seems confusing to have an instance method and a global function both named `ActivateBestChain`. Would suggest renaming one or the other (this is the reason I like lowercase method names) or maybe just dropping the global function since it is only one line.

Reply:
Yeah, this was churn from when I was reverting the inline cs_main locking I was initially doing for the last commit. Reverted, good catch. 

I'm for your ActivateBestChain suggestions, but would prefer to do them in a follow-up PR to limit the scope of this one. If anyone thinks that'd be best done here, I'm happy to.